fix: webhook videocalldata use cal video url

dhairyashiil opened the pull request:
Before:


https://github.com/user-attachments/assets/564b0475-8ab9-4a63-869a-e458688ed174




After:


https://github.com/user-attachments/assets/ed78caa3-e2b5-4315-bf11-89a0ecf5d946



- Fixes #25604 
- Fixes CAL-6869 














<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
videoCallData now uses the Cal.com public video URL for Cal Video (daily_video), so webhook payloads show the canonical link instead of the raw Daily URL. The raw Daily URL is stored as internalProviderUrl and used for BookingReference.meetingUrl to keep embeds working; resolves CAL-6869.

<sup>Written for commit 0db1bbd2672d61b8a3e3ea9a150b5a81be102e2d. Summary will update automatically on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

dhairyashiil committed:
fix: webhook videocalldata use cal video url

dhairyashiil committed:
feat: show Cal.com video URL instead of Daily URL in webhook payload

- Transform daily_video URLs to public Cal.com video URLs (e.g., https://app.cal.com/video/...)
- Updated EventManager.ts for new bookings, reschedules, and location updates
- Updated addVideoCallDataToEvent.ts for reschedule scenarios
- Updated addGuests.handler.ts, connectAndJoin.handler.ts for guest operations
- Updated cancelAttendeeSeat.ts, createNewSeat.ts for seated bookings
- Updated CalendarEventBuilder.ts for calendar event building

Resolves CAL-6869

dhairyashiil committed:
address cubics comment

hariombalhara commented on packages/features/bookings/lib/handleSeats/create/createNewSeat.ts:
```diff
@@ -59,11 +59,13 @@ const createNewSeat = async (
   const videoCallReference = seatedBooking.references.find((reference) => reference.type.includes("_video"));
 
   if (videoCallReference) {
+    const isDailyVideo = videoCallReference.type === "daily_video";
```
Let's not hardcode daily_video string here. I believe there are existing constant or something to know if it is daily video

hariombalhara reviewed (changes requested):
Left 2 comments

dhairyashiil committed:
fix: show Cal.com video URL instead of raw Daily URL in webhook payload

Move Daily video URL transformation to VideoApiAdapter to return public
Cal.com video URL (e.g., https://app.cal.com/video/{uid}) instead of
raw Daily URL (e.g., https://meetco.daily.co/{room}).

Changes:
- VideoApiAdapter now returns getPublicVideoCallUrl() for both regular
  and instant meetings
- Add bookingUid parameter to createInstantCalVideoRoom for URL generation
- Add DAILY_VIDEO_TYPE constant in CalEventParser

Fixes CAL-6869

dhairyashiil committed:
fix: preserve Daily URL for embed while showing Cal.com URL in webhooks

Add providerUrl field to VideoCallData to separate storage URL from display URL.
BookingReference.meetingUrl uses providerUrl (Daily URL) for embed compatibility.

Fixes CAL-6869

pallava-joshi reviewed (commented):
can you check on the failing type checks.

dhairyashiil committed:
fix: exclude providerUrl from webhook videoCallData payload

hariombalhara commented on packages/app-store/dailyvideo/lib/VideoApiAdapter.ts:
```diff
@@ -322,7 +324,7 @@ const DailyVideoApiAdapter = (): VideoApiAdapter => {
   };
 
-  async function createInstantMeeting(endTime: string, region?: RoomGeo) {
+  async function createInstantMeeting(endTime: string, bookingUid: string, region?: RoomGeo) {
```
nit: Better to convert it to name params format. It avoids the issue we are fixing here by moving optional parameter later in the list 
```suggestion
  async function createInstantMeeting({endTime: string, bookingUid: string, region?: RoomGeo)} {
```

hariombalhara commented on packages/features/bookings/lib/EventManager.ts:
```diff
@@ -388,13 +389,16 @@ export default class EventManager {
         thirdPartyRecurringEventId = result.createdEvent?.thirdPartyRecurringEventId;
       }
 
+      const createdEvent = result.createdEvent as { providerUrl?: string; url?: string } | undefined;
```
Instead of this type assertion, can we improve the type of createdEvent itself.

Type assertions are bad for runtime code.

hariombalhara commented on packages/features/bookings/lib/EventManager.ts:
```diff
@@ -447,13 +452,14 @@ export default class EventManager {
-      let event = updatedEvent;
-      if (!event) {
+      let event = (updatedEvent ?? createdEvent) as
+        | { providerUrl?: string; url?: string; id?: string | number; password?: string }
```
Same here. Let's try to remove this type assertion

hariombalhara reviewed (changes requested):
Left some more suggestions. This is a critical change because CalVideo is the default conferencing provider, so we have to be very careful.  
  
Let's think through and manually test each and every flow we touch.  
  
Also, let's also think   
\- Are there any tests that are covering us?   
\- What new tests should we add to avoid regression in the flows we changed

dhairyashiil committed:
fix: use Cal.com URL in webhooks, store Daily.co URL in BookingReference

- Return both public Cal.com URL (url) and internal Daily.co URL (internalProviderUrl) from DailyVideoApiAdapter
- Store internalProviderUrl in BookingReference.meetingUrl for video embed functionality
- Strip internalProviderUrl from evt.videoCallData so webhooks only receive the public URL
- Add getVideoCallData() helper to extract webhook-safe video data
- Add getInternalMeetingProviderUrl() helper to safely extract URL for BookingReference
- Convert createInstantMeeting to use named parameters
- Update InstantBookingCreateService to use internalProviderUrl
- Add tests for new booking and reschedule flows with internalProviderUrl

This ensures:
1. Webhooks receive user-friendly Cal.com URLs (e.g., https://app.cal.com/video/{uid})
2. Video embed page works correctly using the Daily.co URL from BookingReference
3. No breaking changes for other video providers (fallback to url when internalProviderUrl is not present)

Udit-takkar reviewed (changes requested):
The issue is just about not sending the daily video url but here in this PR we are also changing the url that we store in the DB references which would make debugging difficult as after meeting link expire we won't be able to know daily video room name

dhairyashiil committed:
Merge branch 'main' into fix/webhook-videocalldata-use-cal-video-url
